viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -741,6 +744,8 @@ void SLAMServiceImpl::ProcessDataAndStartSavingMaps(double data_start_time) {
     if (!use_live_data) {
         // We still want to optimize the map in localization mode, but we do not
         // need to update the backup of the map
+        cartographer::transform::Rigid3d tmp_global_pose;
```

Comment:
1. Why did we need to move this inside the loop?
2. Is there a risk of us leaking memory from creating new `tmp_global_pose`s every iteration of the while loop without ever freeing the one from the previous loop?

Reply:
1. Because we wanted to keep the function signature simple and not have to pass this var into `AddSensorReading` we wanted to only initialize it if it were going to be used, although now I am getting a little confused why it is better for this to be in the loop. @kkufieta would you mind explaining this again
2. Based on this [doc](https://ptolemy.berkeley.edu/ptolemyclassic/almagest/docs/prog/html/ptlang.doc7.html) I think we actually will need to delete this if we continue with this route

Reply:
This is happening outside the loop. This code is only executed in offline mode after we've processed all the data.